[awesomesjh] iP

awesomesjh opened the pull request.

awesomesjh committed:
Add increment level-0

awesomesjh committed:
Add increment level-1

awesomesjh committed:
Add increment level-2

awesomesjh committed:
Add increment level-3

awesomesjh committed:
Rename Task methods

awesomesjh committed:
Change class variable to local variable

awesomesjh committed:
Add increment level-4

Add support for tracking three types of tasks:
1. Todos: tasks without any date/time attached to it
2. Deadlines: tasks that need to be done before a specific date/time
3. Events: tasks that start at a specific date/time

awesomesjh committed:
Add increment A-TextUiTesting

Use the input/output redirection technique to semi-automate the testing of Jeff

NgYaoDong commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println(divider);
+    }
+
+    public static void main(String[] args) {
```
Avoid Long Methods

ParthGandhiNUS replied:
Perhaps try to use more methods to optimize your main to be a bit more readable

isaaceng7 replied:
Could break it down to more methods

NgYaoDong commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+                    } else if (line.startsWith("deadline ")) {
+                        int byIndex = line.indexOf("/by ");
+                        String description = line.substring(9, byIndex);
+                        String by = line.substring(byIndex + 4);
```
Avoid Magic Numbers

NgYaoDong commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+                    printIndented("Here are the tasks in your list:");
+                    for (int i = 0; i < totalTasks; i++) {
+                        printIndented((i + 1) + "." + tasks[i]);
+                    }
```
Extra semicolon here maybe? Not too sure, unable to see very clearly

NgYaoDong commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+        System.out.println("     " + s);
+    }
+
+    private static void printDivider() {
```
Good job having methods for printing the same things over and over again.

ParthGandhiNUS commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+        printDivider();
+
+        while (true) {
+            String line = in.nextLine();
```
Perhaps you can name the variable "line" as something more intuitive. E.g. userInput or userCommand

ParthGandhiNUS commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+                        String description = line.substring(9, byIndex);
+                        String by = line.substring(byIndex + 4);
+                        tasks[totalTasks] = new Deadline(description, by);
+                    } else {
```
Perhaps can consider refactoring your code to avoid too many nested conditional

ParthGandhiNUS commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.Scanner;
+
+public class Jeff {
```
Nice job for using method for indentation. Perhaps can try to do the same for other processing of user inputs.

isaaceng7 commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+    public static void main(String[] args) {
+        Task[] tasks = new Task[100];
+        int totalTasks = 0;
+        Scanner in = new Scanner(System.in);
```
Could consider using "input" instead of "in"

isaaceng7 commented on src/main/java/Jeff.java:
```diff
@@ -0,0 +1,78 @@
+import java.util.Scanner;
+
+public class Jeff {
+    private static void printIndented(String s) {
```
Good use for creating a method for the indentation.

isaaceng7 replied:
Try to not use past tense, can consider using "printIndentation"

awesomesjh committed:
Refactor code and add error handling

awesomesjh committed:
Update error messages

awesomesjh committed:
Merge branch 'branch-Level-5'

awesomesjh committed:
Move all classes into jeff package

awesomesjh committed:
Merge branch 'branch-A-Packages'

awesomesjh committed:
Add delete functionality

awesomesjh committed:
Add save functionality

Save the tasks in data/jeff.txt when a new task is added.
Load the tasks from data/jeff.txt when the chatbot starts up.
Create data/jeff.txt if the file does not exist.

awesomesjh committed:
Add save functionality for mark/unmark

awesomesjh committed:
Merge branch 'branch-Level-6'

awesomesjh committed:
Merge branch 'branch-Level-7'

# Conflicts:
#	src/main/java/jeff/ExceptionHandler.java

awesomesjh committed:
Minor bug fix

awesomesjh committed:
Refactor exceptions into jeff.exceptions package

awesomesjh committed:
Add save functionality for delete task

awesomesjh committed:
Bug fix

Mark status is now correctly loaded into TaskList

awesomesjh committed:
Create JAR file

sevenseasofbri commented on src/main/java/jeff/CommandHandler.java:
```diff
@@ -0,0 +1,123 @@
+import jeff.exceptions.*;
+
+public class CommandHandler {
+    private static final int TODO_DESCRIPTION_INDEX = 5;
```
good usage of constants

sevenseasofbri commented on src/main/java/jeff/CommandHandler.java:
```diff
@@ -0,0 +1,123 @@
+    }
+
+    public static void handleTodo(String userInput) throws InvalidTodoSyntaxException {
+        if (userInput.equals("todo")) {
```
you can avoid the use of magic literals by storing them as static final vars (constants) or use enums. 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-magic-numbers

sevenseasofbri commented on src/main/java/jeff/CommandHandler.java:
```diff
@@ -0,0 +1,123 @@
+        Printer.printAddTask();
+    }
+
+    public static void handleDeadline(String userInput) throws InvalidDeadlineSyntaxException {
+        int byIndex = userInput.indexOf(" /by ") + 1;
+        if (byIndex == 0) {
+            throw new InvalidDeadlineSyntaxException();
+        }
+        String description = userInput.substring(DEADLINE_DESCRIPTION_INDEX, byIndex);
+        String by = userInput.substring(byIndex + DEADLINE_BY_INDEX_LENGTH);
+        if (description.isEmpty()) {
+            throw new InvalidDeadlineSyntaxException();
+        }
+        Deadline deadline = new Deadline(description, by);
+        TaskList.add(deadline);
+        FileManager.appendTask(deadline);
+        Printer.printAddTask();
+    }
+
+    public static void handleEvent(String userInput) throws InvalidEventSyntaxException {
```
spacing in the functions can be improved.

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#structure-code-logically

sevenseasofbri commented on src/main/java/jeff/CommandHandler.java:
```diff
@@ -0,0 +1,123 @@
+    public static void handleEvent(String userInput) throws InvalidEventSyntaxException {
+        int fromIndex = userInput.indexOf(" /from ") + 1;
+        int toIndex = userInput.indexOf(" /to ") + 1;
+        if (fromIndex == 0 || toIndex == 0 || fromIndex >= toIndex) {
```
good job with practicing 'making the happy path prominent'

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#make-the-happy-path-prominent

sevenseasofbri commented on src/main/java/jeff/CommandHandler.java:
```diff
@@ -0,0 +1,123 @@
+            throw new InvalidDeleteSyntaxException();
+        }
+    }
+
```
the functions above have quite a few things in common, maybe to reduce redundancy, you can abstract out parts that are similar. 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#slap-hard

sevenseasofbri commented on src/main/java/jeff/Printer.java:
```diff
@@ -0,0 +1,64 @@
+    }
+
+    public static void printDivider() {
+        String divider = "    ____________________________________________________________";
```
consider making this a constant

sevenseasofbri commented on src/main/java/jeff/UserInterface.java:
```diff
@@ -0,0 +1,53 @@
+import java.util.Scanner;
+
+public class UserInterface {
+    public static void handleUserInput() {
```
method abstraction is very well done, however it is a little longer than what would be acceptable (>30 LOC) see if you can abstract something out into a different function 

https://nus-cs2113-ay2324s2.github.io/website/se-book-adapted/chapters/codeQuality.html#avoid-long-methods

sevenseasofbri commented on src/main/java/jeff/UserInterface.java:
```diff
@@ -0,0 +1,53 @@
+                ExceptionHandler.handleInvalidUnmarkSyntaxException();
+            } catch (InvalidDeleteSyntaxException e) {
+                ExceptionHandler.handleInvalidDeleteSyntaxException();
+            }
+            finally {
+                Printer.printDivider();
+            }
```
layout should be:
```java
} finally {
    //...
}
```

https://se-education.org/guides/conventions/java/basic.html#layout

sevenseasofbri commented on src/main/java/jeff/UserInterface.java:
```diff
@@ -0,0 +1,53 @@
+package jeff;
+
+import jeff.exceptions.*;
```
avoid wildcard imports

sevenseasofbri reviewed (commented):
Good job 👍🏽

awesomesjh committed:
Replace wildcard imports with single class imports

awesomesjh committed:
Follow coding standard

awesomesjh committed:
Replace magic literals with constants

awesomesjh committed:
Rename classes

awesomesjh committed:
Refactor code

awesomesjh committed:
Add more packages

awesomesjh committed:
Add find functionality

Give users a way to find a task by searching for a keyword

awesomesjh committed:
Store deadline dates as java.time.LocalDate objects

awesomesjh committed:
Add JavaDoc comments to non-private classes/methods

awesomesjh committed:
Merge pull request #1 from awesomesjh/branch-Level-8

Branch level 8

awesomesjh committed:
Merge branch 'master' into branch-Level-9

awesomesjh committed:
Merge pull request #2 from awesomesjh/branch-Level-9

Branch level 9

awesomesjh committed:
Merge branch 'master' into branch-A-JavaDoc

# Conflicts:
#	src/main/java/jeff/ExceptionHandler.java
#	src/main/java/jeff/Printer.java
#	src/main/java/jeff/commands/DeadlineCommand.java
#	src/main/java/jeff/tasks/Deadline.java

awesomesjh committed:
Merge pull request #3 from awesomesjh/branch-A-JavaDoc

Branch a java doc

awesomesjh committed:
Bug fix

Deadline stored in yyyy-mm-dd format no longer throws CorruptFileException when file is read

awesomesjh committed:
Bug fix

Deadline stored in yyyy-mm-dd format no longer throws CorruptFileException when file is read

awesomesjh committed:
Bug fix

Deadline stored in yyyy-mm-dd format no longer throws CorruptFileException when file is read

awesomesjh committed:
Bug fix

Deadline stored in yyyy-mm-dd format no longer throws CorruptFileException when file is read

awesomesjh committed:
Add error checking for invalid characters

awesomesjh committed:
Add a user guide

awesomesjh committed:
Merge branch 'branch-Level-8'

awesomesjh committed:
Merge branch 'branch-Level-9'

awesomesjh committed:
Merge branch 'branch-A-JavaDoc'

awesomesjh committed:
Merge pull request #4 from awesomesjh/branch-A-UserGuide

Add a user guide

awesomesjh committed:
Update user guide

awesomesjh committed:
Modify spacing and indentation in user guide
